python/beeai_framework/tools/mcp_tools.py

```diff
@@ -28,21 +28,22 @@
 
 logger = BeeLogger(__name__)
 
-T = TypeVar("T")
 
-
-@dataclass
-class MCPToolInput:
-    """Input configuration for MCP Tool initialization."""
-
-    client: ClientSession
-    tool: MCPToolInfo
+def json_to_model(model_name: str, schema: dict) -> BaseModel:
+    fields = {}
+    for param_name in schema.get("properties"):
+        fields[param_name] = (
+            str if schema.get("properties").get(param_name).get("type") == "string" else int,
+            Field(description=schema.get("properties").get(param_name).get("description"), default=None),
+        )
+    model = create_model(model_name, **fields)
+    return model
```

Comment:
can this be moved to `beeai_framework/utils/models.py`? it may be useful for others in the future

Reply:
This implementation does not seem very reliable.
Check out this thread https://github.com/pydantic/pydantic/discussions/6598#discussioncomment-11660405